File: Python/Program/Mahasiswa.py
#   Saya Cantika Putri Arbiliansyah dengan NIM 2103727 mengerjakan soal Latihan 4 
#   dalam Praktikum mata kuliah Desain dan Pemrograman Berbasis Objek, untuk keberkahan-Nya
#   maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin

#IMPORT FILE SIVITAS SEBAGAI PARENT
from Sivitas import Sivitas

#KELAS MAHASISWA ADALAH CHILD SIVITAS
class Mahasiswa(Sivitas):
    ## private class
    __nim = ""
    __fakultas = ""

    #KONSTRUKTOR
    def __init__(self, nik="", nama="", gender="", asalUniv = "", email="", nim="", fakultas=""):
        super().__init__(nik, nama, gender, asalUniv, email)
        self.__nim = nim
        self.__fakultas = fakultas

    #GETTER SETTER NIM
    def get_nim(self):
        return self.__nim

    def set_nim(self, nim):
        self.__nim = nim

    #GETTER SETTER FAKULTAS
    def get_fakultas(self):
        return self.__fakultas

    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas

File: CPP/Program/Mahasiswa.cpp
//LIBRARY
#include "Header.hh"

//KONSTRUKTOR
Mahasiswa::Mahasiswa() : Sivitas()
{
    this->nim = "";
    this->fakultas = "";
}

Mahasiswa::Mahasiswa(string NIK, string nama, string gender, string asal_univ, string email, string nim, string fakultas)
          : Sivitas(NIK, nama, gender, asal_univ, email)
{
    this->nim = nim;
    this->fakultas = fakultas;
}

//SETTER DAN GETTER NIM
string Mahasiswa::get_nim()
{
    return this->nim;
}
void Mahasiswa::set_nim(string nim)
{
    this->nim = nim;
}

//SETTER GETTER FAKULTAS
string Mahasiswa::get_fakultas()
{
    return this->fakultas;
}
void Mahasiswa::set_fakultas(string fakultas)
{
    this->fakultas = fakultas;
}

//DESTRUKTOR
Mahasiswa::~Mahasiswa()
{

}